base/task.jl

```diff
@@ -708,29 +708,35 @@ function list_deletefirst!(W::IntrusiveLinkedListSynchronized{T}, t::T) where T
 end
 
 const StickyWorkqueue = IntrusiveLinkedListSynchronized{Task}
-global const Workqueues = [StickyWorkqueue()]
-global const Workqueue = Workqueues[1] # default work queue is thread 1
-function __preinit_threads__()
-    nt = Threads.nthreads()
-    if length(Workqueues) < nt
-        resize!(Workqueues, nt)
-        for i = 2:nt
-            Workqueues[i] = StickyWorkqueue()
+global Workqueues::Vector{StickyWorkqueue} = [StickyWorkqueue()]
+const Workqueues_lock = Threads.SpinLock()
+const Workqueue = Workqueues[1] # default work queue is thread 1 // TODO: deprecate this variable
+
+function workqueue_for(tid::Int)
+    qs = Workqueues
+    if length(qs) >= tid && isassigned(qs, tid)
```

Comment:
Is this `isassigned` call also a performance trap?

Reply:
this is the correct way to call it